### PR TITLE
LP-1642 Change default PDF download document name

### PR DESCRIPTION
**Pull request by harris-irfan**
[LP-1642](https://philanthropyu.atlassian.net/browse/LP-1642)

Change default document name from "certificate" to "PhilanthropyUniversity_[CourseName]" for certificates downloaded as PDF.

### CODE REVIEW

File: openedx/features/student_certificates/views.py [hunk 1]
```diff
@@ -194,8 +195,12 @@ def download_certificate_pdf(request, certificate_uuid):
 
     pdf_document_object = pdfkit.from_string(certificate_image_html, PDFKIT_IMAGE_PATH, pdfkit_options)
 
+    course_display_name = get_course_display_name_by_uuid(certificate_uuid)
+
     response_pdf_certificate = HttpResponse(pdf_document_object, content_type='application/pdf')
-    response_pdf_certificate['Content-Disposition'] = PDF_RESPONSE_HEADER.format(certificate_pdf_name='certificate')
+    response_pdf_certificate['Content-Disposition'] = PDF_RESPONSE_HEADER.format(
+        certificate_pdf_name='PhilanthropyUniversity_{course_name}'.format(
+            course_name=course_display_name.replace(' ', '')))
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/views.py:203]: Let's please simplify this piece of code to make it more readable. 
Make a new constant in constants for certificate pdf name 

`CERTIFICATE_PDF_NAME = 'PhilanthropyUniversity_{display_name}'`

and update these lines with

```
pdf_name = CERTIFICATE_PDF_NAME.format(display_name=course_display_name.replace(' ', ''))
response_pdf_certificate['Content-Disposition'] = PDF_RESPONSE_HEADER.format(certificate_pdf_name=pdf_name)
```

File: openedx/features/student_certificates/helpers.py [hunk 2]
```diff
@@ -74,6 +74,17 @@ def get_certificate_image_url_by_uuid(verify_uuid):
     )
 
 
+def get_course_display_name_by_uuid(verify_uuid):
+    """
+    :param certificate uuid:
+    :return: display name of the course
+    """
+    course_id = GeneratedCertificate.objects.get(verify_uuid=verify_uuid).course_id
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/helpers.py:82]: What if certificate doesn't exist against provided **verify_uuid**?
Please handle it properly.

File: openedx/features/student_certificates/helpers.py [hunk 3]
```diff
@@ -74,6 +75,21 @@ def get_certificate_image_url_by_uuid(verify_uuid):
...
 
+def get_course_display_name_by_uuid(verify_uuid):
+    """
+    :param certificate uuid:
+    :return: display name of the course
+    """
+    try:
+        course_id = GeneratedCertificate.objects.get(verify_uuid=verify_uuid).course_id
+    except GeneratedCertificate.DoesNotExist:
+        raise Exception(CERTIFICATE_NOT_FOUND_AGAINST_UUID_MSG.format(verify_uuid=verify_uuid))
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/helpers.py:86]: Can you please use django shortcut method to get object or 404

**Comment by harris-irfan** [comment]
> Please don't amend your commits,
> In code review it is convenient to review the changes in last commits after the previous review instead of reviewing whole code and finding the changes.

@zeshanarifios since I'm force pushing, the commit history gets deleted and a new single commit is added. I was having issues when I pushed from local branch harris/LP-1642, it added 55 commits, so Zahid bhai and Ahmed suggested to take new branch from develop, add files manually to it and force push that branch to harris/LP-1642 origin. Therefore I'm using that approach every time I need to push a commit to origin. If you have a better solution in mind, please let me know.

**Comment by zeshanarifios** [comment]
> > Please don't amend your commits,
> > In code review it is convenient to review the changes in last commits after the previous review instead of reviewing whole code and finding the changes.
> 
> @zeshanarifios since I'm force pushing, the commit history gets deleted and a new single commit is added. I was having issues when I pushed from local branch harris/LP-1642, it added 55 commits, so Zahid bhai and Ahmed suggested to take new branch from develop, add files manually to it and force push that branch to harris/LP-1642 origin. Therefore I'm using that approach every time I need to push a commit to origin. If you have a better solution in mind, please let me know.



Let's please delete your remote branch (only remote branch), then push your local branch to remote. This will resolve your issue and also clean your remote branch.

**Pull request closed without merging**